packages/payment-processor/src/payment/eth-proxy.ts

```diff
@@ -0,0 +1,68 @@
+import { ContractTransaction, Signer } from 'ethers';
+import { Web3Provider } from 'ethers/providers';
+import { BigNumberish } from 'ethers/utils';
+
+import { ClientTypes, PaymentTypes } from '@requestnetwork/types';
+
+import { ethereumProxyArtifact } from '@requestnetwork/smart-contracts';
+
+import { EthProxyContract } from '../contracts/EthProxyContract';
+
+import { ITransactionOverrides } from './transaction-overrides';
+import {
+  getAmountToPay,
+  getProvider,
+  getRequestPaymentValues,
+  getSigner,
+  validateRequest,
+} from './utils';
+
+/**
+ * Processes a transaction to pay an ETH Request with the proxy contract.
+ * @param request
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function payWithProxyEthInputDataRequest(
```

Comment:
maybe this name is too complex? following ERC20 template, we could name it: 
```suggestion
export async function payEthProxyRequest(
```